### DISCUSSION TITLE
Some extra info on how "Retrieve Context" for AI assist works

### DISCUSSION THREAD

**Discussion by maxim** [zed-industries/zed#9442] [Feature Requests]
In trying to decide whether it's safe or not for me to use "retrieve context" in AI assist panel, I notice that there doesn't seem to be any info at the time of this writing. It would be great to have some details clarified in a form of a doc, or explanatory UI text when clicking the panel button:

1. Which code is being shared? (Is it based on version control tracked files or all files in dir, or something else?)
2. How is it being shared? (Does it become part of every prompt to ChatGPT? Shared once? What effect will it have on tokens consumed?)
3. Nice to have: link to privacy policy. I.e. a link to policies of OpenAI on what happens to data shared with their API, which I imagine should be safer than their chat product.
4. How final is the decision? Once shared, can it ever be "unshared"? If the feature is disabled later, will it remove data from OpenAI, or just stop sending it in the next prompt?
5. Is the data shared immediately upon agreeing to share, or upon the next assist prompt?

**Comment by maxim** [comment]
Just saw that Zed Preview 0.128.0 [release](https://github.com/zed-industries/zed/releases/tag/v0.128.0-pre) includes this:

> We are temporarily removing the semantic index in order to redesign it from scratch.

If that's how ChatGPT context works, this might be a great opportunity to address some of the above questions.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
